feat(LTS): add a new resource to manage transfer (#3288)
huaweicloud/terraform-provider-huaweicloud: docs/resources/lts_transfer.md

---
subcategory: "Log Tank Service (LTS)"
---

# huaweicloud_lts_transfer

> Manages a LTS transfer task resource within HuaweiCloud.  

Reviewer: ```suggestion
Manages an LTS transfer task resource within HuaweiCloud.  
```

Author: ok

      obs_time_zone_id    = "Etc/GMT"
    }
  }
}
```

### Create a DIS transfer task

```hcl
variable "lts_group_id" {}
variable "lts_stream_id" {}
variable "dis_stream_id" {}
variable "dis_stream_name" {}

resource "huaweicloud_lts_transfer" "test" {
  log_group_id = var.lts_group_id

  log_streams {
    log_stream_id = var.lts_stream_id
  }

  log_transfer_info {
    log_transfer_type   = "DIS"
    log_transfer_mode   = "realTime"
    log_storage_format  = "RAW"
    log_transfer_status = "ENABLE"

    log_transfer_detail {
      dis_id   = var.dis_stream_id
      dis_name = var.dis_stream_name
    }
  }
}
```

### Create a DMS transfer task

```hcl
variable "lts_group_id" {}
variable "lts_stream_id" {}
variable "kafaka_instance_id" {}
variable "kafaka_topic" {}

resource "huaweicloud_lts_transfer" "test" {
  log_group_id = var.lts_group_id

  log_streams {
    log_stream_id = var.lts_stream_id
  }

  log_transfer_info {
    log_transfer_type   = "DMS"
    log_transfer_mode   = "realTime"
    log_storage_format  = "RAW"
    log_transfer_status = "ENABLE"

    log_transfer_detail {
      kafka_id    = var.kafaka_instance_id
      kafka_topic = var.kafaka_topic
    }
  }
}
```

> ### Create an delegated OBS transfer task

Reviewer: ```suggestion
### Create a delegated OBS transfer task
```

Author: fixed

* `dis_id` - (Optional, String) DIS stream ID.  
  This parameter is mandatory when you create a DIS transfer task.

* `dis_name` - (Optional, String) DIS stream name.  
  This parameter is mandatory when you create a DIS transfer task.

* `kafka_id` - (Optional, String) Kafka ID.  
  This parameter is mandatory when you create a DMS transfer task.

* `kafka_topic` - (Optional, String) Kafka topic.  
  This parameter is mandatory when you create a DMS transfer task.

  -> Before creating a DMS transfer task, register your Kafka instance with Kafka ID and Kafka topic first.

* `delivery_tags` - (Optional, List) The list of tag fields will be delivered when transferring.  
  This field must contain the following host information: **hostIP**, **hostId**, **hostName**, **pathFile**, and **collectTime**.
  The common fields include **logStreamName**, **regionName**, **logGroupName**, and **projectId**, which are optional.
  The transfer tag: **streamTag**, which is optional.

## Attribute Reference

In addition to all arguments above, the following attributes are exported:

* `id` - The resource ID.

* `log_group_name` - Log group name.  

## Import

> The log transfer task can be imported using the `id`, e.g.

Reviewer: ```suggestion
The LTS transfer task resource can be imported using the `id`, e.g.
```

Author: ok
